tools/pipeline-generator/README.md

```diff
@@ -0,0 +1,241 @@
+# Pipeline Generator
+
+A CLI tool for generating [MONAI Deploy](https://github.com/Project-MONAI/monai-deploy-app-sdk) application pipelines from [MONAI Bundles](https://docs.monai.io/en/stable/bundle_intro.html).
+
+## Features
+
+- List available MONAI models from HuggingFace
+- Generate complete MONAI Deploy applications from HuggingFace models
+- Support for multiple model sources through configuration
+- Automatic bundle download and analysis
+- Template-based code generation with Jinja2
+- Beautiful output formatting with Rich (Python library for rich text and beautiful formatting)
+
+## Platform Requirements
+
+- **Linux/Unix operating systems only**
+- Compatible with MONAI Deploy App SDK platform support
+- Ubuntu 22.04+ recommended (aligned with main SDK requirements)
+
+## Installation
+
+```bash
+# Clone the repository
+cd tools/pipeline-generator/
+
+# Install with uv (no virtualenv needed - uv manages it per command)
+uv pip install -e ".[dev]"
+```
+
+### Running Commands
+
+With uv, you can run commands directly without a prior "install" (pg is the Pipeline Generator command):
+
+```bash
+uv run pg --help
+uv run pg list
+uv run pg gen MONAI/spleen_ct_segmentation --output ./app
+```
+
+## Usage
+
+### Complete Workflow Example
+
+```bash
+# 1. List available models
+uv run pg list
+
+# 2. Generate an application from a model
+uv run pg gen MONAI/spleen_ct_segmentation --output my_app
+
+# 3. Run the application
+uv run pg run my_app --input /path/to/test/data --output ./results
```

Comment:
Please clarify test data type, i.e. nii or DICOM, and limitation wrt DICOM study.